c/tskit/tables.c

```diff
@@ -629,6 +629,81 @@ takeset_string(char **str, tsk_size_t *len, char *new_str, const tsk_size_t new_
     return 0;
 }
 
+static int
+alloc_empty_ragged_column(tsk_size_t num_rows, void **data_col, tsk_size_t **offset_col)
+{
+    int ret = 0;
+
+    *data_col = tsk_malloc(1);
+    *offset_col = tsk_calloc(num_rows + 1, sizeof(tsk_size_t));
+    if (*data_col == NULL || *offset_col == NULL) {
+        ret = TSK_ERR_NO_MEMORY;
+        goto out;
+    }
+out:
+    return ret;
+}
+
+static int
+check_ragged_column(tsk_size_t num_rows, void *data, tsk_size_t *offset)
+{
+    int ret = 0;
+    if ((data == NULL) != (offset == NULL)) {
+        ret = TSK_ERR_BAD_PARAM_VALUE;
+        goto out;
+    }
+    if (data != NULL) {
+        ret = check_offsets(num_rows, offset, 0, false);
+        if (ret != 0) {
+            goto out;
```

Comment:
This can be triggered by constructing bad offsets, can't it?

Reply:
Tests added!